### PR TITLE
panc: allow all relative paths when non-root prefix is used in assignment

### LINKED ISSUE #190: allow relative path that has prefix to start with index
```pan
object template t;

prefix "/a";
"0" = 1;
```
fails with 
```
syntax error [/tmp/t.pan:4.1-4.8]
first term in path must be a key, not an index
```

but this makes no sense, since prefix is a preprocessor, and actually produces (or would produce) `"/a/0" = 1;`

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #190

### CODE REVIEW

File: panc/src/test/java/org/quattor/pan/utils/PathTest.java [hunk 1]
```diff
@@ -128,6 +128,34 @@ public void testInvalidFirstTerm() {
...
+
+	@Test
+	public void testInvalidFirstTermAllowRelative() {
+		List<String> paths = Arrays.asList("alpha:/0/a",
+				"alpha:0/a", "/0/a");
+		for (String s : paths) {
+			try {
+				new Path(s, true);
+				fail("illegal first term with allow_relative did not throw an exception (" + s
+						+ ")");
```
Comment (jrha) [inline panc/src/test/java/org/quattor/pan/utils/PathTest.java]: Why is this on a new line?

File: panc/src/main/java/org/quattor/pan/utils/Path.java [hunk 2]
```diff
@@ -87,11 +87,19 @@ Centre National de la Recherche Scientifique (CNRS).
             .compile("^([^\\{\\}]*+(\\{[^\\{\\}]*\\})?)*$"); //$NON-NLS-1$
 
     /**
-     * Constructor of a path from a String. If the path does not have the
-     * correct syntax, an EvaluationException will be thrown.
+     * Constructor of a path from a String.
```
Comment (jrha) [inline panc/src/main/java/org/quattor/pan/utils/Path.java]: This doesn't throw an exception any more?

File: panc/src/main/java/org/quattor/pan/utils/Path.java [hunk 3]
```diff
@@ -87,11 +87,19 @@ Centre National de la Recherche Scientifique (CNRS).
...
      */
     public Path(String path) throws SyntaxException {
+        this(path, false);
+    }
 
+    /**
+     * Constructor of a path from a String.
+     * allow_all_relative is a boolean indicating that all relative paths are allowed
+     * (e.g. in case of an assigmnent with a prefix).
+     * If the path does not have the correct syntax, an EvaluationException will be thrown.
```
Comment (jrha) [inline panc/src/main/java/org/quattor/pan/utils/Path.java]: Code says `throws SyntaxException`, comment says `EvaluationException`, are they equivalent?

**Comment by jrha** [comment]
@stdweird now has conflicts with master

**Comment by stdweird** [comment]
@jrha conflict resolved and comments addressed

**Pull request merged**